utils/src/main/java/io/ona/kujaku/utils/CoordinateUtils.java

```diff
@@ -1,52 +1,62 @@
 package io.ona.kujaku.utils;
 
 import android.support.annotation.NonNull;
-
 import com.mapbox.mapboxsdk.geometry.LatLng;
-import com.mapbox.mapboxsdk.geometry.LatLngBounds;
 
 /**
  * Created by Ephraim Kigamba - ekigamba@ona.io on 19/12/2017.
  */
-
 public class CoordinateUtils {
 
     /**
      * Checks whether the provided coordinate is within the provided Bounds.
      *
+     * <strong>NOTE: The order in which you provide the bounds does not affect the results</strong>
+     *
      * @param positionInQuestion
-     * @param myMapBounds
+     * @param topLeft
+     * @param topRight
+     * @param bottomLeft
+     * @param bottomRight
      * @return
      */
-    public static boolean isLocationInBounds(@NonNull LatLng positionInQuestion, @NonNull LatLngBounds myMapBounds) {
-        return isLocationInBounds(positionInQuestion, myMapBounds.getLatNorth(), myMapBounds.getLatSouth(), myMapBounds.getLonEast(), myMapBounds.getLonWest());
+    public static boolean isLocationInBounds(@NonNull LatLng positionInQuestion, LatLng topLeft, LatLng topRight, LatLng bottomLeft, LatLng bottomRight) {
+        Point[] bounds = new Point[4];
+        bounds[0] = new Point(topLeft.getLongitude(), topLeft.getLatitude());
+        bounds[1] = new Point(topRight.getLongitude(), topRight.getLatitude());
+        bounds[2] = new Point(bottomLeft.getLongitude(), bottomLeft.getLatitude());
+        bounds[3] = new Point(bottomRight.getLongitude(), bottomRight.getLatitude());
+
+        return contains(bounds, new Point(positionInQuestion.getLongitude(), positionInQuestion.getLatitude()));
     }
 
     /**
-     * Checks whether the provided coordinate is within the provided Bounds.
+     * Return true if the given point is contained inside the boundary.
+     * See: http://www.ecse.rpi.edu/Homepages/wrf/Research/Short_Notes/pnpoly.html
+     * @param needle The point to check
+     * @return true if the point is inside the boundary, false otherwise
      *
-     * @param positionInQuestion
-     * @param latNorth
-     * @param latSouth
-     * @param lonEast
-     * @param lonWest
-     * @return
      */
-    public static boolean isLocationInBounds(@NonNull LatLng positionInQuestion, double latNorth, double latSouth, double lonEast, double lonWest) {
-        return (positionInQuestion.getLatitude() <= latNorth
-                && positionInQuestion.getLatitude() >= latSouth
-                && positionInQuestion.getLongitude() <= lonEast
-                && positionInQuestion.getLongitude() >= lonWest);
+    public static boolean contains(Point[] bounds, Point needle) {
```

Comment:
Why not use [LatLngBounds.contains()](https://developers.google.com/android/reference/com/google/android/gms/maps/model/LatLngBounds.html#contains(com.google.android.gms.maps.model.LatLng))? Less code to maintain.

Reply:
`com.mapbox.mapboxsdk.geometry.LatLngBounds` is also available, if you'd rather go the MapBox way.

Reply:
I would have to test this & read up on it

Reply:
Both contains method are the same and inaccurate. They are not comparable